src/Symfony/Component/Validator/Constraints/EmailValidator.php

```diff
@@ -22,13 +22,22 @@
 class EmailValidator extends ConstraintValidator
 {
     /**
-     * @var bool
+     * @var string
      */
-    private $isStrict;
+    private $defaultProfile;
 
-    public function __construct($strict = false)
+    /**
+     * @param bool   $strict  Deprecated. If the constraint does not define a
+     *                        validation profile, this will determine if
+     *                        rfc-no-warn should be used as the default profile.
+     * @param string $profile If the constraint does not define a validation
+     *                        profile, this will specify which profile to use.
+     */
+    public function __construct($strict = false, $profile = Email::PROFILE_BASIC_REGX)
```

Comment:
This is much better, as my original implementation would have misbehaved in the edge case situation with _both_ config options set (strict_email = TRUE, and email_profile='html5').  Thanks for catching that!